staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -381,15 +412,18 @@ func (e *Store) Create(ctx context.Context, obj runtime.Object, createValidation
 		}
 		return nil, err
 	}
+	// The operation has succeeded.  Call the cleanup function if there is one,
+	// and then make sure the defer doesn't call it again.
+	if cleanup != nil {
```

Comment:
why do we cleanup before we run AfterCreate and Decorator?

Reply:
Update's inner-loop.  I think we want to bound the begin/cleanup to that helper function.  Otherwise we have to call begin on the inside, publish the cleanup method to the outside function and call it there.

We *could* do that, it just seemed simpler to keep it contained.

Reply:
As discussed today, keeping this ordering